libs/designer-ui/src/lib/editor/initializevariable/util.ts

```diff
@@ -107,44 +142,46 @@ export const createVariableEditorSegments = (variables: InitializeVariableProps[
 
   const nodeMap = new Map<string, ValueSegment>();
 
-  const mappedVariables = variables.map((variable) => {
+  const mappedVariableStrings = variables.map((variable) => {
     const name = convertSegmentsToString(variable.name);
     const type = convertSegmentsToString(variable.type);
-
     const isStringType = type === VARIABLE_TYPE.STRING;
+
     const valueSegments = variable.value;
     const isTokenSegment = isTokenValueSegment(valueSegments);
     let value = convertSegmentsToString(valueSegments, nodeMap);
 
     try {
-      if (!isStringType) {
+      if (!isStringType && !isTokenSegment) {
         value = JSON.parse(value);
       }
-    } catch (_error) {
+    } catch {
       // ignore parse errors
     }
 
-    if (value === '') {
-      return `{ "name": "${name}", "type": "${type}" }`;
-    }
-    let valueStr = '';
-    if (isTokenSegment) {
-      if (type === VARIABLE_TYPE.STRING) {
-        valueStr = `"${value}"`;
+    const parts = [`"name": ${JSON.stringify(name)}`, `"type": ${JSON.stringify(type)}`];
+
+    if (value !== '') {
+      if (isTokenSegment) {
+        if (isStringType) {
+          // Clean up multiline token before quoting
+          const cleaned = value.replace(/[\r\n]+/g, '');
+          parts.push(`"value": ${JSON.stringify(cleaned)}`);
+        } else {
+          parts.push(`"value": ${value}`);
+        }
       } else {
-        valueStr = value;
+        // For literals, quote string or stringify object/array/number
+        const valueStr = isStringType ? JSON.stringify(value) : JSON.stringify(value);
```

Comment:
Wait, shouldn't this be - `isStringType ? JSON.stringify(value) : value;

Reply:
ah no, it was something we had previously, where if it was a string type we used a parsed value, but I had changed it later on. I'll move this back to just use JSON.stringify(value) for both